### PR TITLE
Add E2E tests

### PR DESCRIPTION
**Pull request by nikhil-thomas**
Add e2e tests

Add a test document **docs/tests.md**

### CODE REVIEW

File: deploy/role_binding.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ subjects:
 - kind: ServiceAccount
   name: openshift-pipelines-operator
   # Replace this with the namespace the operator is deployed in.
-  namespace: tekton-pipelines
+  namespace: dev-openshift-pipelines-operator
```
Comment (sthaha) [inline deploy/role_binding.yaml]: why has the namespace changed?

Comment (nikhil-thomas) [reply deploy/role_binding.yaml]: Openshift will no let you name a namespace with a string starting with 'openshift'

Comment (nikhil-thomas) [reply deploy/role_binding.yaml]: we need a namespace to run the tests. The operator should have necessary permissions in that namespace. so i set the namespace as `dev-openshift-pipelines-operator` in the role_binding and used it in the run test command
```
operator-sdk test local ./test/e2e --namespace dev-openshift-pipelines-operator --debug
```

Comment (nikhil-thomas) [reply deploy/role_binding.yaml]: while using the operator with OLM the permissions are set from CSV not from role_binding.yaml

Openshift will no let you name a namespace with a string starting with 'openshift'. Hence the `dev-` prefix

**Pull request merged**